third_party/amd/backend/compiler.py

```diff
@@ -148,8 +148,8 @@ def make_llir(src, metadata, options, capability):
                 llvm.link_extern_lib(llvm_mod, path)
         llvm.optimize_module(llvm_mod, llvm.OPTIMIZE_O3)
         # Set kernel attributes
-        kernels = [fn for fn in llvm_mod.get_functions() if fn.has_public_visibility() and not fn.is_declaration()]
-        assert len(kernels) == 1
+        kernels = [fn for fn in llvm_mod.get_functions() if not fn.is_declaration()]
+        # The public kernel should be kernel 0
```

Comment:
is this really an assumption we can make?

Reply:
It was that way in all of the testing I did - which included all of test_core and the tutorials. But you right, perhaps there are scenario that are different. I'll confirm.

Reply:
I have approved and merged, since all the changes are in the AMD module, but keep in mind that if we ever have a pass that changes the order of functions then this will likely break.

Reply:
thanks! This PR also fixes test_if_call as well (same reason). I'll enable that in some future updates.
Also, I'm looking at some other thngs currently that make require me reworking this part, but at least we are working now for this test.